### DISCUSSION TITLE
simplest way to make widgets appear / disappear?

### DISCUSSION THREAD

**Discussion by leemthai** [Textualize/textual#4846] [Q&A]
What I want to try is this: user can click on a column in a DataTable and that will make the relevant 'filter widget' appear underneath the table.

So I will have 10 filter widget, say. But only want one filter widget to appear at a time. And always in the same space.

I'm sure there are several ways to go about this but can someone point me in the direction of the simplest to try first, maybe?

Thanks

**Comment by Zaloog** [comment]
Hi @leemthai 
you can have 1 filter widget with the css property `visibility:hidden;` and on click to a column change that to `visibility:visible;`.
or you define a css class with the above properties and add/remove that class when you want to show/hide the filter.
In addition you have to adjust the selectable filteroptions according to the column you clicked on your table.

Another solution would be to remove and remount the widget at the specific position. But I think the CSS class solution would be cleaner.

Best regards and good luck :)

**Comment by leemthai** [comment]
Thanks dude. Whilst researching 'visibility' property I came across 'display' property which I think is going to the best choice for the job.

**Reply by kyrlon** [reply]
This worked great for changing tabs and disabling the `Footer`.
